packages/dotcom-ui-header/src/__test__/output/component.spec.tsx

```diff
@@ -95,12 +95,12 @@ describe('dotcom-ui-header', () => {
 
       expect(
         container.querySelector(
-          '.o-header__top-column .o-header__top-column--right a[data-trackable="Subscribe"]'
+          '.o-header__top-column.o-header__top-column--right a[data-trackable="Subscribe"]'
         )
       ).not.toBeNull()
       expect(
         container.querySelector(
-          '.o-header__top-column .o-header__top-column--right a[data-trackable="Sign In"]'
+          '.o-header__top-column.o-header__top-column--right a[data-trackable="Sign In"]'
```

Comment:
**Reviewer Note:** I am intrigued by this test, my changes seem to cause it to fail but I can't understand why or how it ever worked 🤔 

`.o-header__top-column .o-header__top-column--right` should have never worked because as far as I can tell they have always been the same element and never a parent/child relationship - which also makes sense because the second class is a modifier so it shouldn't be a child.

Here are places from main prior to any of my changes that show it being the same element.

https://github.com/Financial-Times/dotcom-page-kit/blob/3dadb7b1986ece05d75f38d5a4598cbbd0445bd1/packages/dotcom-ui-header/src/components/top/partials.tsx#L139

https://github.com/Financial-Times/dotcom-page-kit/blob/3dadb7b1986ece05d75f38d5a4598cbbd0445bd1/packages/dotcom-ui-header/src/components/top/partials.tsx#L215

https://github.com/Financial-Times/dotcom-page-kit/blob/3dadb7b1986ece05d75f38d5a4598cbbd0445bd1/packages/dotcom-ui-header/src/components/sticky/partials.tsx#L81


https://github.com/Financial-Times/dotcom-page-kit/blob/3dadb7b1986ece05d75f38d5a4598cbbd0445bd1/packages/dotcom-ui-header/src/components/sticky/partials.tsx#L155

Reply:
Figured it out! The `NavListRightAnonSticky` [component contains a div with the classes](https://github.com/Financial-Times/dotcom-page-kit/blob/c8ec521523337c4b5cbb69c88a79b2099a0bda14/packages/dotcom-ui-header/src/components/sticky/partials.tsx#L82) `o-header__top-column o-header__top-column--right`. This component is rendered within the `TopColumnRightSticky` [component that has a div with the same classes](https://github.com/Financial-Times/dotcom-page-kit/blob/c8ec521523337c4b5cbb69c88a79b2099a0bda14/packages/dotcom-ui-header/src/components/sticky/partials.tsx#L156-L159). That is why these tests were passing.

Reply:
We can see this HTML being rendered in this screenshot:

<img width="1624" alt="image" src="https://github.com/Financial-Times/dotcom-page-kit/assets/893208/96655794-69b0-4d6f-b3b4-911f5f49cbc5">

Reply:
As your new changes reflects the rendered HTML in the Origami component, I am confident in the changes made here and this test now better reflecting the expected markup.

Reply:
Ohh great find. Thanks Joel! Looks like someone added accidental nesting due to the split of the code into granular functions/components